blog/2023-02-20-introducing-the-dhis2-java-sdk.md

```diff
@@ -0,0 +1,174 @@
+---
+slug: 2023/02/introducing-the-dhis2-java-sdk
+title: Introducing the DHIS2 Java SDK
+authors: claudemamo
+tags: [dhis2-java-sdk, integration]
+---
+
+At HISP Centre, we've been engineering numerous run-of–the-mill integration projects where data is exchanged back and forth with DHIS2 through its [Web API](https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-239/introduction.html). Typically, this data is filtered, mapped, transformed, enriched, and routed to or from an application which could be the same DHIS2 instance, a different instance, or even a distinct application altogether. Our programming language of choice for these projects is usually Java. Java has an [excellent ecosystem](https://camel.apache.org/), but in each project, we kept reinventing the wheel in terms of shipping the data out of and into DHIS2. Our productivity was being disturbed with the donkey work of writing code to serialise in-memory objects, deserialise JSON, handle network failures, authenticate, log events, pool remote connections, retry HTTP errors, and so on. We needed a lightweight flexible library that implemented the low-level plumbing. To this end, the DHIS2 Java SDK was born.
```

Comment:
Rephrased this a little with some grammar fixes and removal of "donkey work" to add more clarity.
```suggestion
At HISP Centre, we've been engineering numerous run-of-the-mill integration projects where data is exchanged back and forth with DHIS2 through its [Web API](https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-239/introduction.html). Typically, this data is filtered, mapped, transformed, enriched, and routed to or from an application which could be the same DHIS2 instance, a different instance, or even a distinct application altogether. 

Our programming language of choice for these projects is usually Java. Java has an [excellent ecosystem](https://camel.apache.org/), but in each project, we kept reinventing the wheel in terms of shipping the data out of and into DHIS2. 

Our productivity was being hampered by the repetitive and time-consuming task of writing code for various low-level technical operations such as serializing in-memory objects, de-serializing JSON, handling network failures, authentication, logging events, pooling remote connections, retrying HTTP errors, and more. We needed a lightweight flexible library that implemented the low-level plumbing. To this end, the DHIS2 Java SDK was born.
```